### PR TITLE
Update plots before scraping

### PR DESCRIPTION
**Pull request by alexrockhill**
Reimplements #5310.

@banesullivan I think these changes are much more modest and within the scope of the `pyvista` as opposed to the `pyvistaqt` library. I'm glad we tested it with `qtpy` compatibility so that we know this actually works in the original PR but we don't need to keep that all in the PR just to test one line so I think this should be okay now.

### CODE REVIEW

**Comment by banesullivan** [comment]
github-actions preview

**Comment by banesullivan** [comment]
@ChristosT, since you're the most versed on the scraper at this time (being one of the only other people to modify it), would you please take a quick look at these changes?

File: pyvista/plotting/utilities/sphinx_gallery.py
Comment (banesullivan) [inline pyvista/plotting/utilities/sphinx_gallery.py:59]: Nitpicky, but would you please add a detailed inline comment here about how `app` is specific to pyvistaqt and this is specific to handling that type of plotter (this will help us in the future)

File: pyvista/plotting/utilities/sphinx_gallery.py
Comment (banesullivan) [inline pyvista/plotting/utilities/sphinx_gallery.py:61]: I want to double check that this `.update()` call is necessary or if it should just be in the nest if-block

**Review by banesullivan — CHANGES_REQUESTED** [review]
Just some nitpicky comments, otherwise this looks good! I'd like @ChristosT's feedback before merging since they are also modifying the scraper significantly in #4938

File: pyvista/plotting/utilities/sphinx_gallery.py
Comment (alexrockhill) [reply pyvista/plotting/utilities/sphinx_gallery.py:61]: If it were nested in the if block then it would still solve my issue but if there were an `iren` that isn't associated with `pyvistaqt` then the `update` call would still be helpful. Not sure if that's a common use case but it seems to be logically consistent with the structure of `pyvista` as I understand it.

**Comment by ChristosT** [comment]
It looks good from my point of view. 
It actually makes sense to want to drain the interaction queues before taking a screenshot!

I merged this branch with my changes and built the documentation locally, as far as I checked everything looks good. We just need to be careful when merging because there is a (trivial) merge conflict in `sphinx._gallery.py` but since I already handled it, we can go ahead and merge this first if every other comment is resolved.

**Review by ChristosT — APPROVED** [review]

**Comment by tkoyama010** [comment]
github-actions preview

**Review by banesullivan — APPROVED** [review]

**Comment by tkoyama010** [comment]
@alexrockhill Thank you for your flexibility in responding to our comments. It is a great contribution.

**Comment by alexrockhill** [comment]
Thanks for the reviews everyone

**Pull request merged**
